flink-ml-lib/src/main/java/org/apache/flink/ml/feature/variancethresholdselector/VarianceThresholdSelectorModelData.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.variancethresholdselector;
+
+import org.apache.flink.api.common.serialization.Encoder;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.file.src.reader.SimpleStreamFormat;
+import org.apache.flink.core.fs.FSDataInputStream;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Model data of {@link VarianceThresholdSelectorModel}.
+ *
+ * <p>This class also provides methods to convert model data from Table to a data stream, and
+ * classes to save/load model data.
+ */
+public class VarianceThresholdSelectorModelData {
+
+    public int numOfFeatures;
+    public int[] indices;
```

Comment:
Do you think it would be better to filter the variances according to the threshold in the Model, instead of in the Estimator, and to keep variances of all indices in the model data? This can make users learn the detailed variances computed during the training process, and make one training process applicable to multiple models with just an adjustment to the threshold.

If the current model data would be better, let's divide `VarianceThresholdSelectorParams` and `VarianceThresholdSelectorModelParams`, and make `VARIANCE_THRESHOLD` an exclusive parameter for the estimator.

Reply:
I think there should only be indices in the model data (so do Spark ML and sklearn),  because the model needs to be transformed on different datasets and it has better performance.
Indeed, it's better to divide `VarianceThresholdSelectorParams` and `VarianceThresholdSelectorModelParams`, I'll do it.